### DISCUSSION TITLE
run nicegui (server) on smartphone

### DISCUSSION THREAD

**Discussion by bojh** [zauberzeug/nicegui#3423] [Ideas / Feature Requests]
Working on a network utility, an idea was coming in my mind: What would be the impact to bring/run the nicegui application directly as an App to a smartphone (android/iOS). Means the app contains the nicegui server, and will directly open an browser (wo decoration) as App screen - similiar to the "native" mode?

**Comment by falkoschindler** [comment]
Just recently I've been working on a [project](https://github.com/falkoschindler/sonos_moments) which I'd love to run on the phone without external server. But I couldn't find a feasible way to run a mobile Python server.

**Reply by bojh** [reply]
sonos: yes, understand immediately. :-)
may i should play around with [py2app](https://github.com/ronaldoussoren/py2app). there seems a way for MacOS, Maybe this works also for iPad's .... iOS?
Does not have some experience yet ...

**Reply by falkoschindler** [reply]
Sure. But it probably won't work on iOS:

> Py2app is a project that targets macOS and does not work on other platforms, such as iOS, Linux or Windows.

https://github.com/ronaldoussoren/py2app/blob/487a55e34746fa6eb857de51f2bd0ac03026592d/doc/supported-platforms.rst#L5

I'm not sure about Android though.

**Reply by bojh** [reply]
Oh yes - THX

**Reply by zworkb** [reply]
did you have a look at [Python for Android](https://github.com/kivy/python-for-android) and [kivy-ios](https://github.com/kivy/kivy-ios) ?

**Reply by falkoschindler** [reply]
Only briefly, as far as I remember. I didn't try it out.
